src/core/reference/include/openvino/reference/ceiling.hpp

```diff
@@ -7,13 +7,36 @@
 #include <cmath>
 #include <cstddef>
 
+#include "openvino/reference/copy.hpp"
+#include "openvino/reference/utils/type_util.hpp"
+
 namespace ov {
 namespace reference {
-template <typename T>
-void ceiling(const T* arg, T* out, size_t count) {
-    for (size_t i = 0; i < count; i++) {
-        out[i] = static_cast<T>(std::ceil(arg[i]));
-    }
+
+/**
+ * @brief Reference implementation of Ceiling operator (integral types).
+ *
+ * @param arg     Input pointer to data.
+ * @param out    Output pointer to results.
+ * @param count  Number of elements in input buffer.
+ */
+template <class T, typename std::enable_if<std::is_integral<T>::value>::type* = nullptr>
+void ceiling(const T* arg, T* out, const size_t count) {
+    copy(arg, out, count);
```

Comment:
What about calling `std::copy_n` directly?

Reply:
The intention is to use `reference::copy` in reference implementation to share same implementation (reference::copy is now just wrapper to `std::copy_n`). If this operator will require custom implementation then it can be changed.
There is no difference in binary size the optimization handles well this simple wrapper.